Release/0.0.1 --> master

DanKazuky opened the pull request:
https://github.com/udistrital/necesidades_cliente/issues/306
migraciones adjuntas a los .go

OscarDuarte7 committed:
Update .drone.yml

Commit:
Con @AlaskaRising, se utiliza acceso a base de datos de preproducción para obtener SQL actual de esquema OIKOS y se agrega a carpeta 'SQL'; se agrega gitignore para no subir binarios; se eliminan los binarios generados #3

Commit:
Con @AlaskaRising: se ajusta app.conf de API, para que las variables tengan el estandar establecido. Se modifica main.go a como se trabaja ahora, modificando la conexión a la BD ya que no era correcta. Se crea nuevo servicio get_dependencias_by_id, para obtener todos los hijos de cualquier dependencia. Se agrega querybuilder a consultas relacionadas con dependencias padre y dependencias hijo #4

mariaortiz0401 committed:
Junto con @AlaskaRising, se realiza nueva version de script de SQL de modelo de datos de OIKOS, corrigiendo principalmente temas relacionados con los lineamientos de bases de datos de la oficina #5

mariaortiz0401 committed:
Se agrega imagen del nuevo modelo #5

mariaortiz0401 committed:
:godmode: Con @AlaskaRising se modifica servicio de get_arbol_dependencias a get_dependencias_hijos, para mayor claridad en cuanto a su función. Así mismo, se realiza primera versión del servicio para traer los padres de una dependencia, logrando traer el árbol de abajo hacia arriba #4

Commit:
se comienza optimización de recorridos en el árbol, ya que la forma plantada tarda mucho #4

Commit:
@AlaskaRising; se finaliza la optimización del servicio que obtiene arbol de dependencias hijas logrando que funcione igual al ya planteado, pero de forma más rápida #4

Commit:
Se realiza nueva version de funcion que devuelva dependencias padre de una dependencia dada por medio de función recursiva. Se debe corregir aun la forma en la que muestra #4

Commit:
Con @AlaskaRising, se termina desarrollo de funciones para traer relación entre dependencias tanto de forma ascendente, como descendente. De forma descendente se realiza por medio de árbol y se corrige que no se traia información del primer nodo. De forma ascendente, se realiza por medio de listas, retornando un arreglo de los padres de la dependencia deseada en ese orden. Se eliminan funciones ya no usadas y se elimina servicio de ArbolDependencias, ya que nadie la usaba y se reemplazó por uno de los nuevos

mariaortiz0401 committed:
Con @AlaskaRising, se actualizan modelos acorde a los ajustes realizados al modelos de bases de datos de oikos y se eliminan controladores y modelos de tablas eliminadas del modelo; se actualiza Swagger #6

AlaskaRising committed:
:goat: Junto a @mariaalejandra9404 se agregan archivos migrate y se actualizan .sql e imagen del modelo para despliegue en ambiente Test #6

mariaortiz0401 committed:
Se eliminan controladores y modelos ya no necesarios #6

mariaortiz0401 committed:
Merge branch 'dev' of https://github.com/udistrital/oikos_api into dev

AlaskaRising committed:
:penguin: Con @mariaalejandra9404 Se hace refactor a los controladores, quedan pendientes las funciones que no fueron auto generadas #6:

mariaortiz0401 committed:
Se eliminan rutas de los modelos y controladores eliminados #6

mariaortiz0401 committed:
Con @AlaskaRising, se crea servicio get_facultades_y_proyectos para reemplazar servicio FacultadesConProyectos. Se ubica nuevo servicio en controlador de facultades y se reescribe para que use maps y arroje más rápido el resultado. Se mantienen ambos servicios mientras se hace transición #6

mariaortiz0401 committed:
Se cambia nombre de controlador a proyecto curricular, ya que es más acorde, se cambia servicio a get_all_proyectos_by_facultades @AlaskaRising #6

mariaortiz0401 committed:
Se agrega servicio get_all_proyectos_by_facultad_id para solamente listar los proyectos curriculares de una facultad dada

mariaortiz0401 committed:
Se vuelve a los modelos originales, debido a que el versionamiento que se intentó hacer rompe el API #6

mariaortiz0401 committed:
Se recupera servicio de ArbolDependencias para mantenerlo en version 1 del API. Se agrega segunda conexión a base de datos para probar versionamiento #6

AlaskaRising committed:
:rabbit:  Se cambia tamaño de tabla Dependencias columna nombre  #6

mariaortiz0401 committed:
:facepunch: Con @AlaskaRising, se modifican funciones de GetAll y GetOne para versionar el API, conectándose a nueva versión de base de datos y transformando modelos a la versión antigua #6

mariaortiz0401 committed:
Merge branch 'dev' of https://github.com/udistrital/oikos_api into dev

mariaortiz0401 committed:
Se completan versionamiento de prueba para tabla dependenciaspara funciones PUT, DELETE y POST @AlaskaRising, #6

mariaortiz0401 committed:
se deja solament el controlador de proyecto curricular en versión 2 de namespace #6

mariaortiz0401 committed:
Con @AlaskaRising, se realiza versionamiento de modelo deasignacion_espacio_fisico_dependencia #6

mariaortiz0401 committed:
se añade versionamiento de modelo tipo_uso #6

mariaortiz0401 committed:
se agrega versionamiento de tabla tipo_espacio_fisico y se corrigen errores de tipo_uso

mariaortiz0401 committed:
se agrega versionamiento de tabla tipo_dependencia

mariaortiz0401 committed:
se agrega versionamiento de dependencia_padre #6

mariaortiz0401 committed:
se agrega versionamiento de tabla dependencia_tipo_dependencia #6

mariaortiz0401 committed:
Se agrega versionamiento de espacio_fisico_padre #6

mariaortiz0401 committed:
se agrega versionamiento de espacio_fisico #6

AlaskaRising committed:
Con @AlaskaRising, se hace versionamiento tipo_uso_espacio_fisico #6

AlaskaRising committed:
:umbrella: Se añaden dos tablas nuevas al esquema oikos

AlaskaRising committed:
Se agregan modelos y controladores de nuevas tablas

mariaortiz0401 committed:
Junto con @AlaskaRising, y dada la revisión realizada por Jhon el día jueves, se eliminan campos de Area y Capacidad de lso modelos y controladores relacionados con espacio_fisico. También se eliminan del SQL #6

mariaortiz0401 committed:
Se agrega versionamiento de tabla campo

mariaortiz0401 committed:
Con @AlaskaRising, se añade versionamiento de tabla espacio_fisico_campo #6

mariaortiz0401 committed:
Se agrega ajuste en tablas que manejan estados, para realizar conversiones de strings a booleanos #6

mariaortiz0401 committed:
Se añade refactor de tablas añadidas recientemente #6

AlaskaRising committed:
Se ajusta getAll de dependencia para traer dependencia_tipo_dependencia #6

AlaskaRising committed:
Se ajusta estado en modelo espacio_fisico_padre

AlaskaRising committed:
Se agrega respuesta JSON a todos los servicios GetAll #6

AlaskaRising committed:
Con @AlaskaRising, se terminó la revisión de los endpoint de Post, GetAll y GetOne. Se agrega un condicional en el GetAll donde, si el arreglo está vacío, se devuelve un json nulo, si no, se devuelve el arreglo. #8

mariaortiz0401 committed:
Junto con @AlaskaRising, se evidencia que el servicio PUT de dependencia padre no funciona bien, por lo que corrige servicio #8

mariaortiz0401 committed:
Junto con @AlaskaRising, se evidencia que el servicio PUT de espacio_fisico_padre no funciona bien, por lo que corrige servicio #8

mariaortiz0401 committed:
Junto con @AlaskaRising, se evidencia que el servicio PUT de espacio_fisico no envía correctamente datos, por lo que se corrige. Se corrigen también puts de campo, dependencia_tipo_dependencia y dependencia para que no sobreescriba datos mal #8

mariaortiz0401 committed:
Con @AlaskaRising, se corrigen puts de tipo_dependencia, tipo_espacio_fisico y tipo_uso ya que estaban reemplazando valores de forma incorrecta #8

mariaortiz0401 committed:
Se ajusta put de modelo tipo_uso_espacio_fisico. Se corrigen todos los GetAll con una forma mejor de manejar la respuesta nula, eliminado el if que se estaba usando y usando unos condicionales que ya existian en el beego #8

mariaortiz0401 committed:
Se añade servicio v1/dependencia_padre/tr_dependencia_padre solicitado por Mafe. #10

mariaortiz0401 committed:
Junto con @AlaskaRising, se corrigen servicios de get_dependencias tanto padres como hijos, actualizados a la nueva version del API #8

mariaortiz0401 committed:
@AlaskaRising, se corrige servicio de ArbolDependencias #8

mariaortiz0401 committed:
Se corrigen las consultas de los servicios de proyectosPorFacultad, FacultadesConProyectos, getAllProyectos by facultad y by facultades #8

mariaortiz0401 committed:
Con @AlaskaRising, se realiza primera version de servicio para listar arbol de espacios fisicos, get_espacios_fisicos_hijos_by_id. Enviando el id del espacio físico se obtienen sus hijos #11

mariaortiz0401 committed:
@AlaskaRising, se agrega servicio para listar los padres de un espacio físico dado en get_espacios_fisicos_padres_by_id #11

mariaortiz0401 committed:
Se corrige consulta de proyectos curriculares a nuevo modelo de datos #8

mariaortiz0401 committed:
Se agrega migraciones de tablas restantes #9

mariaortiz0401 committed:
:koala: Se agrega versión 2 de los controladores para permitir versionamiento V2 a nueva base de datos. De esta manera, queda disponible la v1 (versionada para que no arroje errores en los clientes que la usan) y v2, con la nueva estructura de la bd #6

mariaortiz0401 committed:
Se actualiza drone.yml para despliegue en test #9

mariaortiz0401 committed:
Update .drone.yml

mariaortiz0401 committed:
Corrección de merge

mariaortiz0401 committed:
Merge pull request #12 from udistrital/dev

Dev to test

mariaortiz0401 committed:
Correr CI #26

mariaortiz0401 committed:
Merge pull request #13 from udistrital/dev

Correr CI #26

mariaortiz0401 committed:
Se corrige Drone.yml #9

mariaortiz0401 committed:
Merge pull request #14 from udistrital/dev

Se corrige Drone.yml #9

mariaortiz0401 committed:
Se corrigen migraciones para despligue #9

mariaortiz0401 committed:
Merge pull request #15 from udistrital/dev

Se corrigen migraciones para despligue #9

mariaortiz0401 committed:
Se corrigen archivos de migraciones #9

mariaortiz0401 committed:
Merge pull request #16 from udistrital/dev

Se corrigen archivos de migraciones #9

mariaortiz0401 committed:
Se corrigen archivos de migraciones #9

mariaortiz0401 committed:
Merge pull request #17 from udistrital/dev

Se corrigen archivos de migraciones #9

mariaortiz0401 committed:
Se corrige drone.yml #9

mariaortiz0401 committed:
Merge pull request #18 from udistrital/dev

Se corrige drone.yml #9

mariaortiz0401 committed:
Redesplegar #6

mariaortiz0401 committed:
Merge pull request #19 from udistrital/dev

Redesplegar #6

mariaortiz0401 committed:
Se corrige el entrypoint #6

mariaortiz0401 committed:
Merge pull request #20 from udistrital/dev

Se corrige el entrypoint #6

mariaortiz0401 committed:
Se ajusta servicio de espacios fisicos para listar por tipo de espacio fisico #9

mariaortiz0401 committed:
Merge pull request #21 from udistrital/dev

Se ajusta servicio de espacios fisicos para listar por tipo de espaci…

mariaortiz0401 committed:
Redesplegando #9

mariaortiz0401 committed:
Se corrige Drone.yml para redesplegar #9

mariaortiz0401 committed:
Merge pull request #22 from udistrital/dev

Dev to test

mariaortiz0401 committed:
Se añade swagger

mariaortiz0401 committed:
Merge branch 'dev' of https://github.com/udistrital/oikos_api into dev

mariaortiz0401 committed:
Merge pull request #23 from udistrital/dev

Dev

mariaortiz0401 committed:
Update README.md

mariaortiz0401 committed:
Merge pull request #24 from udistrital/dev

Update README.md

mariaortiz0401 committed:
Update .drone.yml

mariaortiz0401 committed:
Merge pull request #25 from udistrital/dev

Update .drone.yml

mariaortiz0401 committed:
Se actualiza imagen del modelo de datos

mariaortiz0401 committed:
Se actualiza imagen del modelo de datos

AlaskaRising committed:
Update README.md

AlaskaRising committed:
Merge pull request #26 from udistrital/dev

Dev

mariaortiz0401 committed:
Se agregan servicios adicionales hechos en V1 a ruta V2

AlaskaRising committed:
Merge pull request #27 from udistrital/dev

Se agregan servicios adicionales hechos en V1 a ruta V2

mariaortiz0401 committed:
@AlaskaRising; se realiza modificación sobre servicio GetAll en tabla asignacion_espacio_fisico_depedencia ya que no traía toda la informacion de espacios físicos y dependencias #28

mariaortiz0401 committed:
Merge pull request #29 from udistrital/dev

@AlaskaRising; se realiza modificación sobre servicio GetAll en tabla…

mariaortiz0401 committed:
@AlaskaRising; se realiza modificación sobre servicio GetAll en tabla asignacion_espacio_fisico_depedencia ya que no traía toda la informacion de espacios físicos y dependencias #28

mariaortiz0401 committed:
Merge pull request #30 from udistrital/dev

@AlaskaRising; se realiza modificación sobre servicio GetAll en tabla…

mariaortiz0401 committed:
@AlaskaRising; ajuste de drone.yml #28

mariaortiz0401 committed:
Merge pull request #31 from udistrital/dev

@AlaskaRising; ajuste de drone.yml #28

blacburn committed:
Update .drone.yml

actualizacion drone.yml

jjvargass committed:
docs:ajuste readme oficial

jjvargass committed:
devops:nuevo pipeline

jjvargass committed:
ajustes conflickto

jjvargass committed:
devops:ajustes en gitignore y limpiesa de carpeta swagger

jjvargass committed:
devops:ajustes para migraciones

jjvargass committed:
Merge pull request #33 from udistrital/develop

Develop

ludwinggsanchez committed:
feat: udistrital/arka_cliente#569. Se genera la migración para agregar las columnas tipo_terreno_id y tipo_edificacion_id a la tabla espacio_fisico.

ludwinggsanchez committed:
feat:  Se actualiza el modelo EspacioFisicoV2.

ludwinggsanchez committed:
feat: Se actualiza Swagger

ludwinggsanchez committed:
feat: Se actualiza modelo y readme

ludwinggsanchez committed:
fix: Down script modified

ludwinggsanchez committed:
fix: Down script modified

corio27 committed:
Merge pull request #34 from udistrital/feat/AgregarCaracteristicasEspacio

Feat/agregar caracteristicas espacio

AlexFBP commented:
Pendiente por generar go.mod/sum y actualizar pipeline de Drone!
- Mover build inmediatamente antes de las migraciones (y despues de test - sonar)
- permitir que fallen las notificaciones
- cambiar gometalinter por otra dependencia no archivada
- Quitar el bloque "workspace" una vez agregados los go.mod/sum
- Agregar y configurar el bloque "trigger" para que no se genere un pipeline en tags

AlexFBP committed:
management: Merge branch 'master' into develop

AlexFBP committed:
devops: pipeline + go.mod/sum | Ver msg del commit -->

- go.mod/sum agregados con go mod init y go mod tidy
  para garantizar la repetitibilidad entre builds
  y habilitar la detección de vulnerabilidades de Dependabot
- Por lo anterior, no se requiere el bloque workspace
- No disparar pipeline cuando se haga push de tags
- Permitir que el step check_commits sea ignorado hasta
  que sea ajustado para que brinde una métrica objetiva
  en los logs. Queda a criterio del lider técnico a cargo
  el permitir código que no cumpla este lineamiento
- step 'go buld' movido despues de testing e inmediatamente
  antes de las migraciones
- Versión de Go subida de 1.14 a 1.18
- gometalinter fue archivado. Se reemplazó por golangci-lint
- Ignorar fallo al notificar por Telegram

AlexFBP committed:
management: .gitignore actualizado

AlexFBP committed:
feat: auditoria + puerto PG (por defecto 5432)

AlexFBP committed:
fix: migraciones + quitar owner/roles

AlexFBP committed:
docs: limpieza sql, agregado SVG para reemplazar PNGs

AlexFBP committed:
docs: agregado template.env

AlexFBP committed:
fix: valores por defecto

AlexFBP committed:
fix: indentaciones/espacios (plugin go-vscode auto) + orden dependencias

AlexFBP committed:
docs: swagger actualizado

AlexFBP committed:
docs: Badge master actualizado

AlexFBP committed:
management: Merge pull request #38 from udistrital/develop

develop --> release

AlexFBP committed:
fix: (devops) bee setup on migrate/rollback

AlexFBP reviewed (commented):
- Se observa que aparentemente en producción, serían nuevos los controladores v2 de la API
- También, que hay que revisar el estado de despliegue en producción, dado que las migraciones (carpeta database) están todas desde cero, en master/producción no existe
- Pendiente por visto bueno de Arquitectura

AlexFBP commented:
Abierto Kanban para sincronización de DBs
https://tuleap.portaloas.udistrital.edu.co/plugins/tracker/?aid=47781

AlexFBP committed:
management: controladores v1 a su propia carpeta

AlexFBP committed:
management: quitar "_v2" a los que quedaron por fuera

AlexFBP committed:
fix: update utils_oas

AlexFBP reviewed (changes requested):
Se observan varios problemas comparando los controladores en pruebas vs producción

| Detalle | Pruebas v1 | Pruebas v2 | Producción (solo v1) | 
|    ---    |      ---            |         ---        |         ---                      |
|Respuesta vacía en controladores que retornan un arreglo de objetos/resultados | `[{}]` (debería ser `[]` no mas) | `[{}]` (debería ser `[]` no mas) | `null` |
| En controladores GET, comportamiento del query=k1:v1,k2:v2,...,kn:vn | falla en la mayoría de casos, sobretodo en los que hay cambio de nombre | sin novedad | sin problemas |
| En controladores GET, comportamiento de fields=field1,field2,...,fieldN | falla en la mayoría de casos, sobretodo en los que hay cambio de nombre | Sin novedad | Sin novedad |
| En controladores GET, comportamiento de limit/offset=(numero) y order=desc,asc,...,etc | ok | ok | ok |
| En controladores GET, comportamiento de sortby=field1,field2,...,fieldN | falla en la mayoría de casos, sobretodo en los que hay cambio de nombre | Sin novedad | Sin novedad |

AlexFBP committed:
fix: (refactor) tipo_uso #36

AlexFBP committed:
fix: (refactor) tipo_dependencia #36

AlexFBP committed:
fix: (refactor) tipo_espacio_fisico #36

AlexFBP committed:
fix: faltante en bc98fbd, 40f6d3d y 990f28 #36

AlexFBP committed:
fix: (refactor) v1 campo #36

AlexFBP committed:
fix: (refactor) v1 espacio_fisico #36

AlexFBP committed:
fix: direccion faltante

AlexFBP committed:
fix: (refactor) v1 espacio_fisico_padre #36

AlexFBP committed:
fix: (refactor) v1 dependencia + _padre + _tipo_dependencia #36

AlexFBP reviewed (approved):
Se requiere #39 antes de aplicar este PR

AlexFBP committed:
management: Merge pull request #39 from udistrital/feature/fixV1Controllers

Feature/fix v1 controllers --> release
